templates/federated_reporting/mrproper.py

```diff
@@ -183,19 +235,42 @@ def main():
         sys.exit(1)
 
     response = api.fr_remote_hubs()
-    for hub in response['hubs']:
-        if hub["role"] != "feeder":
+    if not response["hubs"]:
+        print(
+            "No attached feeders. Please attach at least one feeder hub before running this script."
+        )
+
+    for hub in response["hubs"]:
+        if hub["role"] != "feeder" or hub["target_state"] != "on":
             continue
 
         feeder_hostkey = hub["hostkey"]
+        feeder_hostname = hub["ui_name"]
+        feederApi = NovaApi(
+            api_user="mrproper",
+            api_password=mrproper_password,
+            cert_path=CERT_PATH,
+            hostname=feeder_hostname,
+        )
+        response = feederApi.status()
+        if response["status"] != 200:
+            print(
+                "Unable to get status for feeder {}. Skipping".format(feeder_hostname)
+            )
+            continue
+
+        sql = "SELECT hub_id FROM __hubs WHERE hostkey = '{}'".format(feeder_hostkey)
+        response = api.query(sql)
+        if response["status"] != 200:
+            print("Unable to query for feeder hub_id. Response was {}".format(response))
+        # query API should return one row, [0], and one column, [0], in rows value
+        feeder_hubid = response["rows"][0][0]
 
         sql = """
-SELECT DISTINCT hostkey
+SELECT DISTINCT hosts.hostkey
 FROM hosts
-WHERE EXISTS(
-  SELECT 1 FROM __hubs WHERE hostkey = '{}'
-  AND __hubs.hub_id = hosts.hub_id
-) AND EXISTS(
+WHERE hub_id = '{0}'
```

Comment:
Much better, thanks!

Reply:
Yep. Good exercise to refactor several times actually. Thanks!